google-cloud-spanner/src/test/java/com/google/cloud/spanner/MockSpannerServiceImpl.java

```diff
@@ -1080,6 +1094,9 @@ private void returnResultSet(
       metadata = metadata.toBuilder().setTransaction(transaction).build();
     }
     resultSet = resultSet.toBuilder().setMetadata(metadata).build();
+    if (session.getMultiplexed()) {
+      resultSet = resultSet.toBuilder().setPrecommitToken(getResultSetPrecommitToken()).build();
+    }
```

Comment:
nit: could we maybe combine these so that we don't execute `toBuilder().build()` too many times?

Reply:
Done